### PR TITLE
Controlling high blood pressure

### PR DESCRIPTION
**Pull request by deepsonshrestha**
## Describe your changes
Please include a summary of any changes.

## How has this been tested?
Please describe the tests you ran to verify your changes.  Provide instructions or code to reproduce output.

## Reviewer focus
Please summarize the specific items you’d like the reviewer(s) to look into.

## Checklist before requesting a review
- [ ] I have updated the version number in dbt_project.yml file to reflect the release number of this PR
- [ ] I have updated the docs files (by running dbt docs generate/serve and copying the necessary files into the docs folder)
- [ ] I have commented my code as necessary
- [ ] I have added at least one Github label to this PR
- [ ] My code follows style guidelines
- [ ] (Optional) I have recorded a Loom to explain this PR

### CODE REVIEW

File: models/quality_measures/intermediate/cqm236_controlling_blood_pressure/quality_measures__int_cqm236_denominator.sql [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+        , code_system
+    from {{ ref('quality_measures__value_sets') }}
+    where lower(concept_name) in (
+          'office visit'
+        , 'home healthcare services'
+        , 'preventive care services established office visit, 18 and up'
+        , 'preventive care services initial office visit, 18 and up'
+        , 'annual wellness visit'
+        , 'telephone visits'
+        , 'nutrition services'
```
Comment (sarahmcmorgan) [inline models/quality_measures/intermediate/cqm236_controlling_blood_pressure/quality_measures__int_cqm236_denominator.sql]: I don't see this concept listed as applicable for this measure. I looked at the concept seed file and in the measure specification. Am I missing something?

File: models/quality_measures/intermediate/cqm236_controlling_blood_pressure/quality_measures__int_cqm236_denominator.sql [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+        , floor({{ datediff('birth_date', 'e.max_date', 'hour') }} / 8760.0) as max_age
+        , qualifying_types
+    from {{ref('quality_measures__stg_core__patient')}} p
+    inner join encounters_by_patient e
+        on p.patient_id = e.patient_id
+    where p.death_date is null
+
+)
+
+, qualifying_patients as (
```
Comment (sarahmcmorgan) [inline models/quality_measures/intermediate/cqm236_controlling_blood_pressure/quality_measures__int_cqm236_denominator.sql:159]: Can you add a comment with the denominator logic above this CTE? I think it helps when someone is looking at the code to understand why a 6 month filter is being added. Here's the text from the measure specification:

> Patients 18-85 years of age who had a visit and diagnosis of essential hypertension starting before and continuing into, or starting during the first six months of the measurement period.

File: models/quality_measures/intermediate/cqm236_controlling_blood_pressure/quality_measures__int_cqm236_exclude_advanced_illness.sql [hunk 3]
```diff
@@ -0,0 +1,327 @@
...
+with patients_with_frailty as (
+
+    select
+          patient_id
+        , performance_period_begin
+        , performance_period_end
+        , exclusion_date
+        , exclusion_reason
+    from {{ ref('quality_measures__int_cqm236__frailty') }}
+    where is_patient_older_than_80 = 0
```
Comment (sarahmcmorgan) [inline models/quality_measures/intermediate/cqm236_controlling_blood_pressure/quality_measures__int_cqm236_exclude_advanced_illness.sql:16]: I really like your solution for using an older than 80 flag. 🌟

File: models/quality_measures/intermediate/cqm236_controlling_blood_pressure/quality_measures__int_cqm236_exclude_hospice_palliative.sql [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+        , case code_system
+            when 'SNOMEDCT' then 'snomed-ct'
+            when 'ICD9CM' then 'icd-9-cm'
+            when 'ICD10CM' then 'icd-10-cm'
+            when 'CPT' then 'hcpcs'
+            when 'ICD10PCS' then 'icd-10-pcs'
+          else lower(code_system) end as code_system
+        , concept_name
+    From {{ref('quality_measures__value_sets')}}
+    where lower(concept_name) in  (
```
Comment (sarahmcmorgan) [inline models/quality_measures/intermediate/cqm236_controlling_blood_pressure/quality_measures__int_cqm236_exclude_hospice_palliative.sql:20]: I think we need to include the concept "Palliative Care Intervention". It's listed in the concept seed for being included in this measure.

File: models/quality_measures/intermediate/cqm236_controlling_blood_pressure/quality_measures__int_cqm236_numerator.sql [hunk 5]
```diff
@@ -0,0 +1,187 @@
...
+        )
+        and normalized_code not in ( --yet to be confirmed by Sarah
+              '99473'
+            , '99474'
+        )
+        -- 99473 
+        -- Self-measured blood pressure using a device validated for clinical accuracy; patient education/training and device calibration
+        -- 99474
+        -- Separate self-measurements of two readings one minute apart, twice daily over a 30-day period (minimum of 12 readings), collection of data reported by the patient and/or caregiver to the physician or other qualified health care professional, with report of average systolic and diastolic pressures and subsequent communication of a treatment plan to the patient
+
```
Comment (sarahmcmorgan) [inline models/quality_measures/intermediate/cqm236_controlling_blood_pressure/quality_measures__int_cqm236_numerator.sql]: These exclusions make sense to me. 👍

File: models/quality_measures/intermediate/cqm236_controlling_blood_pressure/quality_measures__int_cqm236_numerator.sql [hunk 6]
```diff
@@ -0,0 +1,187 @@
...
+            and least_systolic_bp_per_day.observation_date = least_diastolic_bp_per_day.observation_date
+
+)
+
+, numerator as (
+
+    select
+          *
+        , case
+            when systolic_bp < 140 and diastolic_bp < 110 --needs to be changed back to 90; 110 for testing against synthetic data
```
Comment (sarahmcmorgan) [inline models/quality_measures/intermediate/cqm236_controlling_blood_pressure/quality_measures__int_cqm236_numerator.sql]: I will work on updating some observation values in the synthetic dataset tomorrow morning to help with this issue. Thanks for bringing it to my attention. 🙏

**Review by sarahmcmorgan — COMMENTED** [review]
Thanks for all of your work on this! It's looking really good. Just left a couple of comments/requests.

File: models/quality_measures/intermediate/cqm236_controlling_blood_pressure/quality_measures__int_cqm236_denominator.sql (hunk 1 above)
Comment (renish-maitri) [reply models/quality_measures/intermediate/cqm236_controlling_blood_pressure/quality_measures__int_cqm236_denominator.sql]: Apart from nutrition services (_which came in due to merge issue at our end_), we can see the above concepts in value_set_concepts table using the query below
`select * from QUALITY_MEASURES_SEEDS.QUALITY_MEASURES._value_set_concepts where measure_id='CQM236'`

File: models/quality_measures/intermediate/cqm236_controlling_blood_pressure/quality_measures__int_cqm236__frailty.sql [hunk 7]
```diff
@@ -0,0 +1,264 @@
+{{ config(
+     enabled = var('quality_measures_enabled',var('claims_enabled',var('clinical_enabled',var('tuva_marts_enabled',False))))
+   )
```
Comment (sarahmcmorgan) [inline models/quality_measures/intermediate/cqm236_controlling_blood_pressure/quality_measures__int_cqm236__frailty.sql]: Hey @deepsonshrestha, can you add an ` | as_bool` to the config logic for all of your new models? You can take a look at the final models to see how it works. We recently found a bug and had to add this fix.

Comment (deepsonshrestha) [reply models/quality_measures/intermediate/cqm236_controlling_blood_pressure/quality_measures__int_cqm236__frailty.sql]: `| as bool` has been added to all the config models

**Pull request merged**